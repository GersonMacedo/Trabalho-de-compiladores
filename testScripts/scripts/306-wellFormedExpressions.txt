! -s CONTENT --disable-logs printer
! should accept correct expressions
program p ;
    var b1 : boolean;
    var b2 : boolean;
    var i1 : integer;
    var i2 : integer;
begin
    i1 := 99;
    i1 := i2;
    i1 := (i2);
    i1 := 1 + 2;
    i1 := 3 * 4;
    i1 := 5 / 6;

    b1 := true;
    b1 := false;
    b1 := b2;
    b1 := (b2);
    b1 := 1 < 2;
    b1 := 3 > 4;
    b1 := 5 = 6;
    b1 := true = false;
    b1 := true and false;
    b1 := false or true;

    i1 := ((i2 + 3) * (i1 - 4) / 5) + 42;
    b1 := ((5 < 6) and (4 > 3) or (2 = 3)) and b2;
end .