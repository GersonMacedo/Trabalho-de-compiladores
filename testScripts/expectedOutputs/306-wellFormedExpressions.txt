
Performing up to content analysis at testScripts/scripts/306-wellFormedExpressions.txt
No errors found
